based/crates/reth/src/unsealed_block.rs

```diff
@@ -94,28 +111,40 @@ impl UnsealedBlock {
         })
     }
 
-    /// Decoded txs (allocates Vec), like Go `Transactions()` but decoded
+    /// Decoded txs (allocates Vec), like Go `Transactions()` but decoded.
     pub fn transactions(&self) -> Result<Vec<TxEnvelope>, UnsealedBlockError> {
         self.transactions_iter_decoded().collect()
     }
 
-    /// Raw tx bytes (allocates Vec<Vec<u8>>), like Go `ByteTransactions()`
+    /// Raw tx bytes (allocates Vec<Vec<u8>>), like Go `ByteTransactions()`.
     pub fn byte_transactions(&self) -> Vec<Vec<u8>> {
         self.transactions_iter_bytes().map(|tx| tx.iter().copied().collect::<Vec<u8>>()).collect()
     }
 
-    // Return the last frag on the list.
+    /// Return the last fragment in the list (if any).
     pub fn last_frag(&self) -> Option<&FragV0> {
         self.frags.last()
     }
 
     /// Apply the accepted frag into in-memory bookkeeping (NOT executing txs).
     ///
     /// Execution results (receipts/logs/gas) should be appended separately.
-    pub fn accept_frag(&mut self, f: FragV0) {
+    pub fn accept_frag_execution(
+        &mut self,
+        f: FragV0,
+        logs: Vec<Log>,
+        receipts: Vec<TransactionReceipt<OpReceiptEnvelope<Log>>>,
+        cummulative_gas_used: u64,
+    ) {
         self.last_sequence_number = Some(f.seq);
         self.cumulative_blob_gas_used = self.cumulative_blob_gas_used.saturating_add(f.blob_gas_used);
-        self.frags.push(f);
+        self.frags.push(f.clone());
+        self.logs.extend_from_slice(logs.as_slice());
+        self.receipts.extend_from_slice(receipts.as_slice());
+        self.cumulative_gas_used = cummulative_gas_used;
+
+        // TODO: Is this correct? Is everything applied here?
```

Comment:
most is applied here, transaction list and db cache are applied separately but called before this